profiles/serializers_test.py

```diff
@@ -303,3 +311,117 @@ def test_delete_employment(self):
 
         # Other profile is unaffected
         assert employment3.profile.work_history.count() == 1
+
+
+class ProfileFilledOutTests(ESTestCase):
+    """Tests for validating filled out profiles"""
+
+    @classmethod
+    def setUpTestData(cls):
+        """
+        Create a profile and social auth
+        """
+        with mute_signals(post_save):
+            cls.profile = ProfileFactory.create(
+                filled_out=True,
+                agreed_to_terms_of_service=True,
+            )
+        EmploymentFactory.create(profile=cls.profile)
+        EducationFactory.create(profile=cls.profile)
+        cls.profile.user.social_auth.create(
+            provider=EdxOrgOAuth2.name,
+            uid="{}_edx".format(cls.profile.user.username)
+        )
+
+    def setUp(self):
+        """
+        Create a profile and social auth
+        """
+        serializer = ProfileFilledOutSerializer(self.profile)
+        self.data = serializer.to_representation(self.profile)
+
+    def test_success(self):
+        """
+        Test a successful validation
+        """
+        ProfileFilledOutSerializer(data=self.data).is_valid(raise_exception=True)
+
+    def test_filled_out_false(self):
+        """
+        filled_out cannot be set to false
+        """
+        self.data['filled_out'] = False
+        with self.assertRaises(ValidationError) as ex:
+            ProfileFilledOutSerializer(data=self.data).is_valid(raise_exception=True)
+        assert ex.exception.args[0] == {'non_field_errors': ['filled_out cannot be set to false']}
+
+    def test_agreed_to_terms_of_service_false(self):
+        """
+        agreed_to_terms_of_service cannot be set to false
+        """
+        self.data['agreed_to_terms_of_service'] = False
+        with self.assertRaises(ValidationError) as ex:
+            ProfileFilledOutSerializer(data=self.data).is_valid(raise_exception=True)
+        assert ex.exception.args[0] == {'non_field_errors': ['agreed_to_terms_of_service cannot be set to false']}
+
+    def assert_required_fields(self, field_names, parent_getter, field_parent_getter):
+        """
+        Helper function to assert required fields
+        """
+        for key in field_names:
+            field = field_parent_getter(ProfileFilledOutSerializer().fields)[key]
+            # skip fields that are generated, read only, or that tie to other serializers which are tested elsewhere
+            if isinstance(field, (ListSerializer, SerializerMethodField, ReadOnlyField)):
+                continue
+
+            clone = deepcopy(self.data)
+            parent_getter(clone)[key] = None
+            with self.assertRaises(ValidationError) as ex:
+                ProfileFilledOutSerializer(data=clone).is_valid(raise_exception=True)
+            assert parent_getter(ex.exception.args[0]) == {key: ['This field may not be null.']}
+
+            field = field_parent_getter(ProfileFilledOutSerializer(data=clone).fields)[key]
```

Comment:
1. can you move this function under `setUp` rather than mixed in among the test cases?
1. i don't think this line is necessary. the definition on line 372 seems to be taking care of this. i commented this line out and the tests succeeded